packages/solutions/app-tools/src/config/default.ts

```diff
@@ -1,23 +1,26 @@
-import { createDefaultConfig as createDefaultBuilderConfig } from '@modern-js/builder-webpack-provider';
 import { MAIN_ENTRY_NAME } from '@modern-js/utils';
 import { getAutoInjectEnv } from '../utils/env';
 import { IAppContext, AppUserConfig, AppLegacyUserConfig } from '../types';
 
+// Define some default values that are different from rsbuild default config or used in useResolvedConfigContext
 export function createDefaultConfig(
   appContext: IAppContext,
-  bundler: 'webpack' | 'rspack',
 ): AppUserConfig<'webpack'> | AppUserConfig<'rspack'> {
-  const defaultBuilderConfig = createDefaultBuilderConfig();
-
   const dev: AppUserConfig['dev'] = {
-    ...defaultBuilderConfig.dev,
     // `dev.port` should not have a default value
     // because we will use `server.port` by default
     port: undefined,
   };
 
   const output: AppUserConfig['output'] = {
-    ...defaultBuilderConfig.output,
+    distPath: {
```

Comment:
why do we need to re-define default config here

Reply:
api.useResolvedConfigContext requires it. api.useResolvedConfigContext needs to be set & used regardless of whether builder config is init / modify or not. 
I don't think this is a good design, but it's a bit hard to change...